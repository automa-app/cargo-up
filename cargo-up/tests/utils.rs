use assert_cmd::Command;
use insta::assert_snapshot;
use std::{
    fs::{read_to_string, write},
    path::PathBuf,
    str::from_utf8,
};

#[allow(dead_code)]
pub fn run(dir: &str, args: &[&str]) -> (String, String) {
    let output = Command::cargo_bin("cargo-up")
        .unwrap()
        .current_dir(dir)
        .args(args)
        .output()
        .unwrap();

    let out = from_utf8(&output.stdout).unwrap();
    let err = from_utf8(&output.stderr).unwrap();

    (out.to_string(), err.to_string())
}

#[allow(dead_code)]
pub fn run_out(dir: &str, args: &[&str]) -> String {
    let (out, err) = run(dir, args);

    assert!(err.is_empty());
    out
}

#[allow(dead_code)]
pub fn run_err(dir: &str, args: &[&str]) -> String {
    let (out, err) = run(dir, args);

    assert!(out.is_empty());
    err
}

#[allow(dead_code)]
pub fn run_upgrader(dir: &str, version: &str) -> (String, String) {
    let mut fixture_on = PathBuf::new();

    fixture_on.push("..");
    fixture_on.push("fixtures");
    fixture_on.push(dir);
    fixture_on.push("on");

    let on = fixture_on.clone();

    fixture_on.push("src");
    fixture_on.push("main.rs");

    let original = read_to_string(&fixture_on).unwrap();

    let (out, err) = run(
        &on.to_string_lossy(),
        &[
            "up",
            "dep",
            "upgradee",
            "--lib-path",
            "../../../cargo-up",
            "--path",
            "../../upgrader",
            "--name",
            "upgrader",
            "--to-version",
            version,
            "--suppress-cargo-output",
        ],
    );

    let file_content = read_to_string(&fixture_on).unwrap();
    assert_snapshot!(file_content);

    write(&fixture_on, original).unwrap();

    (out, err)
}
