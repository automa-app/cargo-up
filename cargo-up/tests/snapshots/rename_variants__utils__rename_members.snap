---
source: tests/utils.rs
expression: file_content
---
use upgradee::Enum;

type E = Enum;

fn main() {
    let a = Enum::Pineapple;
    s(&a);

    let b = E::Pineapple;
    t(&b);

    if let Enum::Pineapple = b {}

    let mut e = Enum::Berry(8);

    if let Enum::Berry(g) = e.clone() {}
    if let Enum::Berry(ref mut g) = e {}

    let mut f = Enum::Papaya { size: 10 };

    if let Enum::Papaya { size } = e.clone() {}
}

fn s(v: &Enum) -> Option<bool> {
    use Enum::*;

    match v {
        Orange => Some(true),
        Grape(_) => Some(false),
        Melon { .. } => Some(false),
        _ => None,
    }
}

fn t(v: &Enum) -> Option<bool> {
    use Enum::*;

    if let Orange = v {
        Some(true)
    } else if let Grape(g) = v {
        Some(false)
    } else if let Melon { size } = v {
        Some(false)
    } else {
        None
    }
}

